dvc/progress.py

```diff
@@ -33,62 +35,84 @@ class Tqdm(tqdm):
         """
         bytes   : shortcut for
             `unit='B', unit_scale=True, unit_divisor=1024, miniters=1`
-        desc_truncate  : like `desc` but will truncate to 10 chars
+        desc  : persists after `close()`
+        level  : effective logging level for determining `disable`;
+            used only if `disable` is unspecified
         kwargs  : anything accepted by `tqdm.tqdm()`
         """
-        kwargs = deepcopy(kwargs)
+        kwargs = kwargs.copy()
+        kwargs.setdefault("unit_scale", True)
         if bytes:
-            for k, v in dict(
-                unit="B", unit_scale=True, unit_divisor=1024, miniters=1
-            ).items():
-                kwargs.setdefault(k, v)
-        if desc_truncate is not None:
-            kwargs.setdefault("desc", self.truncate(desc_truncate))
-        if disable is None:
-            disable = (
-                logging.getLogger(__name__).getEffectiveLevel()
-                >= logging.CRITICAL
+            kwargs = merge(
+                dict(unit="B", unit_scale=True, unit_divisor=1024, miniters=1),
+                kwargs,
             )
+        self.desc_persist = desc
+        if disable is None:
+            disable = logger.getEffectiveLevel() > level
         super(Tqdm, self).__init__(
-            iterable=iterable, disable=disable, leave=leave, **kwargs
+            iterable=iterable,
+            disable=disable,
+            leave=leave,
+            desc=desc,
+            bar_format="!",
+            **kwargs
         )
+        if bar_format is None:
+            if self.__len__():
+                self.bar_format = self.BAR_FMT_DEFAULT
+            else:
+                self.bar_format = self.BAR_FMT_NOTOTAL
+        else:
+            self.bar_format = bar_format
+        self.refresh()
 
-    def update_desc(self, desc, n=1, truncate=True):
+    def update_desc(self, desc, n=1):
         """
-        Calls `set_description(truncate(desc))` and `update(n)`
+        Calls `set_description_str(desc)` and `update(n)`
         """
-        self.set_description(
-            self.truncate(desc) if truncate else desc, refresh=False
-        )
+        self.set_description_str(desc, refresh=False)
         self.update(n)
 
     def update_to(self, current, total=None):
         if total:
             self.total = total  # pylint: disable=W0613,W0201
         self.update(current - self.n)
 
-    @classmethod
-    def truncate(cls, s, max_len=25, end=True, fill="..."):
-        """
-        Guarantee len(output) < max_lenself.
-        >>> truncate("hello", 4)
-        '...o'
-        """
-        if len(s) <= max_len:
-            return s
-        if len(fill) > max_len:
-            return fill[-max_len:] if end else fill[:max_len]
-        i = max_len - len(fill)
-        return (fill + s[-i:]) if end else (s[:i] + fill)
+    def close(self):
+        if self.desc_persist is not None:
+            self.set_description_str(self.desc_persist, refresh=False)
```

Comment:
Does this make sense if `leave=False`?

Reply:
should be fine this way as

1. `refresh=False` so it won't trigger a potentially costly update
2. the string representation of the bar (possibly useful for debugging purposes and/or future use) will be updated